File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    public int solution(int[] numbers, int target) {
        Queue<Integer> q = new LinkedList<>();
        int answer = 0;
        q.offer(0);
        for(int i = 0; i < numbers.length; i++){
            int size = q.size();
            for(int j = 0; j < size; j++){
                int sum = q.poll();
                q.offer(sum + numbers[i]);
                q.offer(sum - numbers[i]);
            }
        }
        while(!q.isEmpty()){
            if(q.poll() == target) answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>
#define endl '\n'
#define loop(i, s, n) for(int i = s; i <= n; i++)
#define MAXN 21

using namespace std;

int bfs(vector<int> numbers, int target){
    int cnt = 0;
    queue<int> q; q.push(0);
    loop(i, 0, numbers.size() - 1){
        int size = q.size();
        loop(j, 0, size - 1){
            int t = q.front(); q.pop();
            q.push(t + numbers[i]);
            q.push(t - numbers[i]);
        }
    }
    while(!q.empty()){
        if(q.front() == target) cnt++;
        q.pop();
    }
    return cnt;
}

int solution(vector<int> numbers, int target) {
    return bfs(numbers, target);
}